recipes/folly/all/conanfile.py

```diff
@@ -44,44 +45,32 @@ def config_options(self):
             tools.check_min_cppstd(self, self._minimum_cpp_standard)
-        min_version = self._minimum_compilers_version.get(
-            str(self.settings.compiler))
+        min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
         if not min_version:
-            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
-                self.name, self.settings.compiler))
+            self.output.warn("{} recipe lacks information about the {} compiler support.".format(self.name, self.settings.compiler))
         else:
             if tools.Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration("{} requires C++14 support. The current compiler {} {} does not support it.".format(
-                    self.name, self.settings.compiler, self.settings.compiler.version))
+                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
+                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
 
         if self.settings.os == "Windows" and self.settings.arch != "x86_64":
-            raise ConanInvalidConfiguration("Folly requires a 64bit target architecture")
-        elif self.settings.os == "Windows" and self.settings.compiler == "Visual Studio" and \
-                "MT" in self.settings.compiler.runtime:
-            raise ConanInvalidConfiguration("Folly could not be build with runtime MT")
-        elif self.settings.os == "Macos" and self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built by apple-clang as shared library")
-        elif self.settings.os == "Windows" and self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on Windows as a shared library")
-        elif Version(self.version) >= "2020.08.10.00" and self.settings.compiler == "Visual Studio" and \
-                not self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built on Windows as a static library")
-        elif Version(self.version) >= "2020.08.10.00" and self.settings.compiler == "clang" and \
-                self.options.shared:
-            raise ConanInvalidConfiguration("Folly could not be built by clang as a shared library")
+            raise ConanInvalidConfiguration("Folly requires a 64bit target architecture on Windows")
 
-        self._strip_options_requirements()
+        if self.settings.os in ["Macos", "Windows"] and self.options.shared:
+            raise ConanInvalidConfiguration("Folly could not be built on {} as shared library".format(self.settings.os))
 
-    def _strip_options_requirements(self):
-        self.options["boost"].header_only = False
-        for boost_comp in self._required_boost_components:
-            setattr(self.options["boost"], "without_{}".format(boost_comp), False)
```

Comment:
Why is it removed?

Reply:
It looked like a duplicate, it had no effect so I removed it

Reply:
It's not a duplicate. Should boost recipe modifies its default options, this recipe would still work out of the box.

Reply:
🤔 I am not in favor of that... It would be missing the correct boost package_id from the options and would not build CI would just fail =/

Reply:
CI is one thing, some people don't use pre build binaries, they build from source or just download CCI recipes for their own repository. A recipe should work out of the box with its default options.